mlrun/runtimes/pod.py

```diff
@@ -1201,6 +1181,91 @@ def with_requests(
         """
         self.spec.with_requests(mem, cpu, patch=patch)
 
+    def detect_preemptible_node_selector(self, node_selector: dict[str, str]) -> None:
+        """
+        Checks if any provided node selector matches the preemptible node selectors.
+        Issues a warning if a selector may be pruned at runtime depending on preemption mode.
+
+        :param node_selector: The user-provided node selector dictionary.
+        """
+        preemptible_node_selector = mlconf.get_preemptible_node_selector()
+
+        for key, value in node_selector.items():
+            if preemptible_node_selector.get(key) == value:
+                self.raise_preemptible_warning(
+                    message=f"Node selector '{key}: {value}' may be removed at runtime"
+                )
+
+    def detect_preemptible_tolerations(
+        self, tolerations: list[k8s_client.V1Toleration]
+    ) -> None:
+        """
+        Checks if any provided toleration matches preemptible tolerations.
+        Issues a warning if a toleration may be pruned at runtime depending on preemption mode.
+
+        :param tolerations: The user-provided list of tolerations.
+        """
+        preemptible_tolerations = [
+            k8s_client.V1Toleration(
+                key=toleration.get("key"),
+                value=toleration.get("value"),
+                effect=toleration.get("effect"),
+            )
+            for toleration in mlconf.get_preemptible_tolerations()
+        ]
+
+        transform_attribute_to_k8s_class_instance("tolerations", tolerations)
```

Comment:
why is this needed and you can't just use `mlconf.get_preemptible_tolerations()`?

Reply:
The response of mlconf.get_preemptible_tolerations() is a dictionary, and I wanted to compare the V1Toleration objects, but maybe the `transform_attribute_to_k8s_class_instance("tolerations", tolerations)` can be removed, since the `self.spec.tolerations` runs it prior to this function